code/API_definitions/Optimal-Edge-Discovery.yaml

```diff
@@ -0,0 +1,741 @@
+############################################################################
+servers:
+  - url: "http://localhost:{port}/v1"
+    description: Local host API
+    variables:
+      port:
+        enum:
+          - "3333"
+        default: "3333"
+  - url: "https://edge.example.com:{port}/v1"
+    description: Production API
+    variables:
+      port:
+        enum:
+          - "443"
+          - "8443"
+        default: "443"
+############################################################################
+#                                     Tags                                 #
+############################################################################
+tags:
+  - name: Regions
+    description: |
+      Get all Region IDs and Names for edge cloud zones
+  - name: Discovery
+    description: |
+      Discover the regions and zones in the edge cloud
+      application, find optimal edge cloud zones for
+      your deployed applications, and optimal
+      Application Endpoints for your clients to connect to.
+############################################################################
+#                                     Paths                                #
+############################################################################
+paths:
+  /regions:
+    get:
+      summary: Fetch all Regions
+      operationId: get-regions
+      parameters:
+        - name: x-correlator
+          in: header
+          required: false
+          description: |
+            When the API Consumer includes the "x-correlator" header in the
+            request, the API provider must include it in the response with
+            the same value that was used in the request. Otherwise, it is
+            optional to include the "x-correlator" header in the response with
+              any valid value. Recommendation is to use UUID for values.
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Returns Edge Cloud Regions
+          content:
+            application/json:
+              schema:
+                title: GetEdgeCloudRegionsResponse
+                type: array
+                items:
+                  $ref: "#/components/schemas/EdgeCloudRegions"
+        "400":
+          $ref: "#/components/responses/Generic400"
+        "401":
+          $ref: "#/components/responses/Generic401"
+        "403":
+          $ref: "#/components/responses/Generic403"
+        "404":
+          $ref: "#/components/responses/Generic404"
+        "406":
+          $ref: "#/components/responses/Generic406"
+        "422":
+          $ref: "#/components/responses/Generic422"
+        "429":
+          $ref: "#/components/responses/Generic429"
+      tags:
+        - Regions
+      description: |
+        This is a helper API to get all the regions. The response identifies the
+        regions where edge cloud zones are available.
+  /edge-cloud-zones:
```

Comment:
```suggestion
  /retrieve-optimal-edge-cloud-zone:
```

Reply:
^ this change is to align with Simple Edge Discovery (retrieve-closest-edge-cloud-zone) and to follow [API design guidelines](https://github.com/camaraproject/Commonalities/blob/main/documentation/CAMARA-API-Design-Guide.md#65-post-or-get-for-transferring-sensitive-or-complex-data):

"When the POST method is used:

the resource in the path MUST be a verb (e.g. retrieve-location and not location) to differentiate from an actual resource creation"